applications/nrf_desktop/src/util/chmap_filter/include/chmap_filter.h

```diff
@@ -0,0 +1,255 @@
+#define DEFAULT_PARAM_WIFI_ACTIVE_THRESHOLD    75   /* = 0.75 */
+#define DEFAULT_PARAM_WIFI_RATING_TRIM         50   /* = 0.5 */
+#define DEFAULT_PARAM_EVAL_SUCCESS_THRESHOLD   85   /* = 0.85 */
+
+/**@brief Channel map filter parameters that can be changed at runtime
+ *
+ * @note Type 'fix' is used to indicate a fixed point value with a scaling
+ * factor of 1/100. E.g. value 20 = 20/100 = 0.2
+ * @note Threshold values are ratios of average channel rating.
+ * E.g. PARAM_BLE_BLOCK_THRESHOLD = 20 will block a BLE channel if its rating
+ * falls below 20% of the average rating.
+ * @note The time unit is based on how often @ref chmap_filter_process is run.
+ * Recommend running this function at ~1 second interval
+ */
+struct chmap_filter_params {
+/* Maintenance algorithm needs these many samples to run [int] */
+	u16_t maintenance_sample_count;
+/* Initial channel rating [int] */
+	s32_t initial_rating;
+/* Min num of channels in channel map [int] */
+	u8_t  min_channel_count;
+/* Channel rating weight of CRC OK [fix] */
+	s16_t ble_weight_crc_ok;
+/* Channel rating weight of CRC ERR [fix] */
+	s16_t ble_weight_crc_error;
+/* Factor for BLE channel rating trim. Must be < 100. */
+/* Note: "high" value can lead to rating overflow [fix] */
+	s16_t ble_rating_trim;
+/* Threshold of average rating required to block of single channel [fix] */
+	s16_t ble_block_threshold;
+/* BLE channel suspected of being affected by wifi will not be blocked as*/
+/* single channel until this time has passed [int] */
+	u16_t ble_wifi_keepout_duration;
+/* Wifi strength increase factor. Higher value blocks wifi faster [fix] */
+	s16_t wifi_rating_inc;
+/* Threshold of average rating that indicates Wifi presence [fix] */
+	s16_t wifi_present_threshold;
+/* Threshold of average rating required to block  wifi channel [fix] */
+	s16_t wifi_active_threshold;
+/* Factor for Wifi channel rating trim. Must be < 100. [fix] */
+	s16_t wifi_rating_trim;
+/* Maximum number of channels that can be evaluated at a time [int]*/
+	u8_t  eval_max_count;
+/* Evaluated channels will be evaluated for these many time units [int] */
+	u16_t eval_duration;
+/* Blocked channels will not be evaluated for these many time units [int]*/
+	u16_t eval_keepout_duration;
+/* Rating of evaluated channel must be above this threshold [fix] */
+	s16_t eval_success_threshold;
+/* Dynamic block and evaluation durations. [bool] */
+/* Fewer channels in channel map = shorter time until evaluation. */
+/* Channel blocked in the past = longer time until evaluation */
+	bool dynamic_durations;
+};
+
+struct chmap_instance;
+
+/**@brief Get library version number
+ *
+ * @return Pointer to version string
+ */
+const char *chmap_filter_version(void);
+
+/**@brief Initialize channel map filter state
+ *
+ * @details This library version is using a single filter instance,
+ * stored internally
+ *
+ * @return pointer to channel map instance
+ */
+struct chmap_instance *chmap_filter_init(void);
```

Comment:
Yes, that's right. Can refactor this to use supplied memory to hold state information instead of hidden variable if you think this is a better approach.